pennylane/ops/op_math/sum.py

```diff
@@ -458,3 +470,460 @@ def get_summands(self, cutoff=1.0e-12):
+                    "different sets of wires"
+                )
+
+            coeffs2 = other.coeffs
+            ops2 = other.ops
+
+            coeffs = qml.math.kron(coeffs1, coeffs2)
+            ops_list = itertools.product(ops1, ops2)
+            terms = [qml.prod(t[0], t[1], lazy=False) for t in ops_list]
+            return qml.LinearCombination(coeffs, terms)
+
+        if isinstance(other, Operator):
+            if other.arithmetic_depth == 0:
+                new_ops = [op @ other for op in self.ops]
+
+                # build new pauli rep using old pauli rep
+                if (pr1 := self.pauli_rep) is not None and (pr2 := other.pauli_rep) is not None:
+                    new_pr = pr1 @ pr2
+                else:
+                    new_pr = None
+                return LinearCombination(self.coeffs, new_ops, _pauli_rep=new_pr)
+            return qml.prod(self, other)
+
+        return NotImplemented
+
+    def __add__(self, H):
+        r"""The addition operation between a LinearCombination and a LinearCombination/Tensor/Observable."""
+        ops = self.ops.copy()
+        self_coeffs = copy(self.coeffs)
+
+        if isinstance(H, numbers.Number) and H == 0:
+            return self
+
+        if isinstance(H, (LinearCombination, qml.Hamiltonian)):
+            coeffs = qml.math.concatenate([self_coeffs, copy(H.coeffs)], axis=0)
+            ops.extend(H.ops.copy())
+            if (pr1 := self.pauli_rep) is not None and (pr2 := H.pauli_rep) is not None:
+                _pauli_rep = pr1 + pr2
+            else:
+                _pauli_rep = None
+            return qml.LinearCombination(coeffs, ops, _pauli_rep=_pauli_rep)
+
+        if isinstance(H, (Tensor, Observable, qml.ops.Prod, qml.ops.SProd)):
+            coeffs = qml.math.concatenate(
+                [self_coeffs, qml.math.cast_like([1.0], self_coeffs)], axis=0
+            )
+            ops.append(H)
+
+            return qml.LinearCombination(coeffs, ops)
+
+        return NotImplemented
+
+    __radd__ = __add__
+
+    def __mul__(self, a):
+        r"""The scalar multiplication operation between a scalar and a LinearCombination."""
+        if isinstance(a, (int, float)):
+            self_coeffs = copy(self.coeffs)
+            coeffs = qml.math.multiply(a, self_coeffs)
+            return qml.LinearCombination(coeffs, self.ops.copy())
+
+        return NotImplemented
+
+    __rmul__ = __mul__
+
+    def __sub__(self, H):
+        r"""The subtraction operation between a LinearCombination and a LinearCombination/Tensor/Observable."""
+        if isinstance(H, (LinearCombination, qml.Hamiltonian, Tensor, Observable)):
+            return self + qml.s_prod(-1.0, H, lazy=False)
+        return NotImplemented
+
+    def queue(self, context=qml.QueuingManager):
+        """Queues a qml.LinearCombination instance"""
+        for o in self.ops:
+            context.remove(o)
+        context.append(self)
+        return self
```

Comment:
Why is this done in the original Hamiltonian class and is there a nicer way to do this now?
(simply removing it raises a lot of errors)